lib/fs_utils/generate.js

```diff
@@ -85,7 +84,7 @@ const concat = (files, path, definitionFn, autoRequire, config) => {
   const processor = (file) => {
     root.add(file.node);
     const data = file.node.isIdentity ? file.data : file.source;
-    if (isJs && data.trim().substr(-1) !== semi) root.add(semi);
+    if (isJs && !data.trim().endsWith(';')) root.add(';');
```

Comment:
Relatively short strings will be cached by v8. `trim` can be improved: we are slicing whitespaces from both sides, but only interested in last `\S` char. I will replace it with regexp: patterns with anchors are very fast.